ui/perfherder/ReplicatesGraph.jsx

```diff
@@ -0,0 +1,157 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { react2angular } from 'react2angular/index.es2015';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faCog } from '@fortawesome/free-solid-svg-icons';
+
+import perf from '../js/perf';
+import { errorMessageClass } from '../helpers/constants';
+import ErrorBoundary from '../shared/ErrorBoundary';
+import Graph from '../shared/Graph';
+import PerfSeriesModel from '../models/perfSeries';
+import { getData } from '../helpers/http';
+import { createApiUrl, perfSummaryEndpoint } from '../helpers/url';
+
+// TODO remove $stateParams after switching to react router
+export default class ReplicatesGraph extends React.Component {
+  // TODO: sync parent with children IRT dataLoading
+  constructor(props) {
+    super(props);
+
+    this.state = { dataLoading: true };
+  }
+
+  async componentDidMount() {
+    this.syncGraphData().then(() => {
+      window.document.title = `${this.state.platform}: ${this.state.testName}`;
+      this.setState({ dataLoading: false });
+    });
+  }
+
+  syncGraphData = async () => {
+    const { title } = this.props;
+
+    const { replicateData, drawingData } = await this.fetchReplicateGraphData();
+    const { replicateDataError } = replicateData;
+    const numRuns = drawingData ? drawingData.numRuns : 0;
+
+    return this.setState({
+      drawingData,
+      graphSpecs: this.getGraphSpecs(title, numRuns, replicateDataError),
+    });
```

Comment:
You can do this lines 26-27 as a callback in setState, which ensures the state you've just updated is available after its updated:
```
    return this.setState({
      drawingData,
      graphSpecs: this.getGraphSpecs(title, numRuns, replicateDataError,
      loading: false,
     }, () =>
      window.document.title = `${this.state.platform}: ${this.state.testName});
```